final-project/helpers.py

```diff
@@ -198,3 +198,54 @@ def find_product(product_requested):
 
     # Otherwise, return information about the requested product.
     return product_data
+
+
+def get_reference(product_with_references):
+    """Search for references to product research."""
+
+    # If no name is given, then ask the user for one.
+    if not product_with_references:
+        return apology("Please specify product")
+    elif product_with_references == None:
+        return apology("Please specify product")
+
+    # Search for references to a given product
+    references_for_product = db.execute("""
+            SELECT research.link, title
+            FROM research
+            LEFT JOIN products
+            ON products.id=research.product_id
+            WHERE products.product_name=:product
+            """, product=product_with_references)
+
+    # Check whether we found references
+    if not references_for_product:
+        return apology("We have no references. Please help us research")
+    elif references_for_product == None:
+        return apology("We have no references. Please help us research")
+
+    # Save the number of research articles for this products.
+    number_of_references = len(references_for_product)
+
+    # Save the titles and links to each reference.
+    # Create an empty list to save our titles.
+    reference_titles = []
+    # Create an empty list to save our links.
+    reference_links = []
+
+    # Loop through each title in our list of references
+    for reference in range(number_of_references):
```

Comment:
Why not just `for reference in references_for_product`?

Reply:
Good idea.

For some reason in the template I cannot use Jinga to loop through a list with `for reference in references_for_product`. I am not sure why. I adopted the habit of using range within the Python code as well. However, you're right, I do not have any reason to do that here. Good spot.

Reply:
Wow that code looks so much nicer 👍